### PR TITLE
Remove instance types and preferences

### PR DESCRIPTION
**Pull request by akrejcir**
**What this PR does / why we need it**:
SSP operator should not longer create `Instancetypes` and `Preferences`. This PR removes the code that creates them and adds code to clean up any existing `Instancetypes` and `Preferences` that were created before this PR.

**Which issue(s) this PR fixes**: 
Jira: https://issues.redhat.com/browse/CNV-44448

**Release note**:
```release-note
SSP no longer creates instance types and preferences.
```

### CODE REVIEW

**Review by 0xFelix — COMMENTED** [review]
Thinking about it, I'm not even sure we need a cleanup. virt-operator takes ownership of resources previously owned by others, granted they are present in the bundle that virt-operator is going to deploy. This would allow us to drop any instancetypes or preferences related code from SSP.

https://github.com/kubevirt/kubevirt/blob/6fb17773c7dddb9c9a526959605eb5acf07619cc/tests/operator/operator.go#L2573

What do you think?

**Comment by akrejcir** [comment]
> Thinking about it, I'm not even sure we need a cleanup. virt-operator takes ownership of resources previously owned by others, granted they are present in the bundle that virt-operator is going to deploy. This would allow us to drop any instancetypes or preferences related code from SSP.
> 
> https://github.com/kubevirt/kubevirt/blob/6fb17773c7dddb9c9a526959605eb5acf07619cc/tests/operator/operator.go#L2573
> 
> What do you think?

I would leave the cleanup code here. There may be a case, where the bundle in virt-operator does not contain the same resources as the bundle in SSP. Or maybe the resources were created from a URL, and the virt-operator uses different URL.

Related to this, I think I should add code to remove SSP owner annotations from instance types and preferences that are owned by virt-operator. So we would still need some instance-types related code.

**Comment by 0xFelix** [comment]
Can we sync with @lyarwood what SSP should do? We really should keep it as simple as possible.

**Comment by lyarwood** [comment]
> > Thinking about it, I'm not even sure we need a cleanup. virt-operator takes ownership of resources previously owned by others, granted they are present in the bundle that virt-operator is going to deploy. This would allow us to drop any instancetypes or preferences related code from SSP.
> > https://github.com/kubevirt/kubevirt/blob/6fb17773c7dddb9c9a526959605eb5acf07619cc/tests/operator/operator.go#L2573
> > What do you think?
> 
> I would leave the cleanup code here. There may be a case, where the bundle in virt-operator does not contain the same resources as the bundle in SSP. Or maybe the resources were created from a URL, and the virt-operator uses different URL.

I'm fine leaving this clean up code in place for a release but after that I think this can itself be removed in v0.23.0.
 
> Related to this, I think I should add code to remove SSP owner annotations from instance types and preferences that are owned by virt-operator. So we would still need some instance-types related code.

I thought virt-operator already updated all of the ownership metadata of these resources? 

If it's missing something that's a bug in virt-operator right?

**Comment by akrejcir** [comment]
> > Related to this, I think I should add code to remove SSP owner annotations from instance types and preferences that are owned by virt-operator. So we would still need some instance-types related code.
> 
> I thought virt-operator already updated all of the ownership metadata of these resources?
> 
> If it's missing something that's a bug in virt-operator right?

I've read the `virt-operator` code that creates common instance types and preferences. I see that it overwrites annotations and labels, so any annotations that were previously on the existing object are removed.
This means that users cannot add their own annotations and labels to these objects, is that what we want?

If yes, then it simplifies this PR, because `virt-operator` will remove the owner annotations that point to SSP.

**Comment by lyarwood** [comment]
> > > Related to this, I think I should add code to remove SSP owner annotations from instance types and preferences that are owned by virt-operator. So we would still need some instance-types related code.
> > 
> > 
> > I thought virt-operator already updated all of the ownership metadata of these resources?
> > If it's missing something that's a bug in virt-operator right?
> 
> I've read the `virt-operator` code that creates common instance types and preferences. I see that it overwrites annotations and labels, so any annotations that were previously on the existing object are removed. This means that users cannot add their own annotations and labels to these objects, is that what we want?
> 
> If yes, then it simplifies this PR, because `virt-operator` will remove the owner annotations that point to SSP.

Yes I believe this is valid as these resources are solely managed by virt-operator.

**Comment by akrejcir** [comment]
Ok, then I don't need to change this PR.

/unhold

**Review by 0xFelix — COMMENTED** [review]
The automation updating the common-instancetype bundles also needs to be taken care of. Can you please have a look at it?

/approve

File: tests/common_instancetypes_test.go [hunk 1]
```diff
@@ -1,146 +1,105 @@
...
+				Memory: instancetypev1beta1.MemoryInstancetype{
+					Guest: resource.MustParse("1G"),
+				},
+			},
+		}
+
+		Expect(apiClient.Create(ctx, instanceType)).To(Succeed())
+		DeferCleanup(func() {
+			Expect(apiClient.Delete(ctx, instanceType)).To(Or(Succeed(), MatchError(errors.IsNotFound, "errors.IsNotFound")))
 		})
```
[the remark this answers is not in the document]
Comment (akrejcir) [reply tests/common_instancetypes_test.go:64]: It is good if tests cases cleanup even if they fail, else the changes in the cluster may affect other test cases.

File: tests/common_instancetypes_test.go [hunk 2]
```diff
@@ -1,146 +1,105 @@
 package tests
 
 import (
+	"time"
+
 	. "github.com/onsi/ginkgo/v2"
 	. "github.com/onsi/gomega"
+	"k8s.io/apimachinery/pkg/types"
+	ssp "kubevirt.io/ssp-operator/api/v1beta2"
```
[the remark this answers is not in the document]
Comment (akrejcir) [reply tests/common_instancetypes_test.go:9]: Thanks. I don't want to retrigger CI only for this code style change.

**Pull request merged**